articles/150_roslaunch.md

```diff
@@ -0,0 +1,170 @@
+
+In addition, the launch system may interact with, or allow the user to interact with, an operating system process's:
+
+- `stdin` pipe
+- signals (`SIGINT`, `SIGTERM`, `SIGUSR1`, etc...)
+
+Regardless of how the user uses the launch system to interact with these items, they should be exposed by the launch system, which is the only entity which can interact with them directly.
+
+##### Termination
+
+If the operating system process terminates, and therefore returns a return code, the launch system will report this event and it can be handled in a user defined way.
+Termination covers expected termination (e.g. return from `main()` or use `exit()`) and unexpected termination (e.g. the abort trap or a segmentation fault or bus error).
+
+Historically, ROS 1's `roslaunch` allowed a few common exit handling cases:
+
+- `require=true`: if this process exits (any reason) shutdown everything else, as it's "required"
+- `respawn=true`: if this process exits (any reason) restart it with the same settings as startup
+  - `respawn_delay=N`: if restarting it, delay a number of seconds between attempts
+
+#### Shell Evaluation
+
+A special case of operating system processes, shell evaluation would simply be passing shell script code as an argument to the default system shell.
+This is a problematic thing to support because it is hard/messy to make it portable to all operating systems.
+
+An in between entity is an operating system process which uses shell evaluation to expand a relative executable name to an absolute path using the PATH environment variable.
+
+#### Remote Operating System Processes
+
+<div class="alert alert-warning" markdown="1">
+TODO: Describe this as a "mix-in" which can convert anything based on an "operating system process" into a remote one by adding ssh/user/setup/etc... information needed to do so, see how ROS 1's roslaunch does it.
+</div>
+
+#### Process with a Single Node
+
+This is a ROS specific entity which contains a single ROS node.
+This was likely the most commonly used type of entity launched in ROS 1, as you could only have one node per process in ROS 1.
+In ROS 2, this will likely be less common, but will still be used quite a bit in the form of quickly developed scripts and drivers or GUI tools which might require control over the main thread.
+
+In 
+
+#### Process with Multiple Nodes
+
+#### Dynamically loaded Nodes
+
+##### By Configuration
+
+##### By Proxy
+
+### System Description
+
+<div class="alert alert-warning" markdown="1">
+TODO: Restructure notes on this and put them here.
+
+Temporary summary:
+
+This is the section where the Python API in `ros2/launch` and the XML file format from ROS 1 overlap.
+Basically how do you describe the system in a way that is flexible, but also verifiable and ideally can also be statically analyzed (at least for something like the XML format).
+
+I have compare and contrast like notes for other systems like upstart, systemd, and launchd. 
```

Comment:
> Is there some thought of integrating with your capabilities package at some point? Does that even make sense?

I don't know, I'll have to think about it more, but I think it would be good to mention in this document, if only to better describe how the launch system is different from a system like that.

I think it makes sense to consider it, but I don't know off hand if it makes sense to integrate something like the capabilities server into the launch system, or to instead build it on top.